### PR TITLE
Fix The requested qty is not available

### PR DESCRIPTION
**Pull request by SergeyP18**
#1637 
### Description

Start to create order from admin and try to add created to 'Items Ordered' out of stock product by sku
Two error messages are displayed :
'The requested qty is not available' - unnecessary,
'This product is out of stock.' - is correct
### Fixed Issues (if relevant)
I check the total balance in the warehouses and after that I spend the rest of the checks
### Manual testing scenarios
Start to create order from admin and try to add created to 'Items Ordered' out of stock product by sku
### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1911) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1911) before we can accept your contribution.<br/><hr/>**Sergey** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by smoskaluk** [comment]
@SergeyP18 The problem still reproducing with product assigned to default source, please recheck.

File: app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php [hunk 1]
```diff
@@ -84,13 +84,22 @@ public function execute(string $sku, int $stockId, float $requestedQty): Product
             return $this->productSalableResultFactory->create(['errors' => $errors]);
         }
 
+        if(!(bool)$stockItemData['is_salable']){
+            $stockItemData['quantity'] = 0;
```
Comment (maghamed) [inline app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php:88]: for consistency you need to use constants:
`GetStockItemDataInterface::QUANTITY` for `quantity` and 
`GetStockItemDataInterface::IS_SALABLE` for `is_salable`

File: app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php [hunk 2]
```diff
@@ -84,13 +84,22 @@ public function execute(string $sku, int $stockId, float $requestedQty): Product
...
+        }
```
Comment (maghamed) [inline app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php:89]: it's better not to nullify the quantity of `$stockItemData[GetStockItemDataInterface::QUANTITY]` (ideally this array should be read-only) if the product is not salable but still proceed with computations 

```
        /** @var StockItemConfigurationInterface $stockItemConfiguration */
        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);	        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
         $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +	        $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +
            $this->getReservationsQuantity->execute($sku, $stockId);	            $this->getReservationsQuantity->execute($sku, $stockId);
        $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;	        $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;
```

but instead to add conditional logic here:
```
        if ((bool)$stockItemData[GetStockItemDataInterface::IS_SALABLE]) {
            $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +
                $this->getReservationsQuantity->execute($sku, $stockId);
            $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;
            $isEnoughQty = $qtyLeftInStock >= 0;
        } else {
            $isEnoughQty = false;
        }
```

**Review by maghamed — CHANGES_REQUESTED** [review]

**Comment by maghamed** [comment]
merged into for proceeding with further changes

**Pull request merged**